src/api/common/HttpWrapper.ts

```diff
@@ -0,0 +1,81 @@
+import axios, {
+  AxiosError,
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosResponse,
+} from 'axios';
+
+const LOGMAXLEN = 120;
+const DEFAULT_TIMEOUT = 30000;
+
+const stringify = (something: unknown): string =>
+  JSON.stringify(something, null, 3);
+
+const interceptors = {
+  logRequest: {
+    onResponse: (response: AxiosResponse) => {
+      console.log(
+        `Got a response: ${response.config.url as string}${stringify(
+          response.data
+        ).slice(0, LOGMAXLEN)}`
+      );
+
+      return response;
+    },
+    onError: (error: AxiosError) => {
+      const { response, message, config } = error;
+      if (error.request) {
+        console.log(`Request failed: ${stringify(error.request)}`);
+      }
+      if (response) {
+        console.log(`Response error: ${stringify(response)}`);
+      }
+      if (!error.request && !response) {
+        console.log('Error while setting up request', stringify(message));
+      }
+      console.log(`Configuration: ${stringify(config)}`);
+
+      return Promise.reject(error);
+    },
+  },
+};
+
+export default class HttpWrapper {
+  private readonly _httpClient: AxiosInstance;
+
+  private readonly _isLogEnabled: boolean;
+
+  constructor(opts: AxiosRequestConfig = {}, isLogEnabled = false) {
+    this._httpClient = axios.create({
+      timeout: DEFAULT_TIMEOUT,
+      headers: { 'Content-Type': 'application/json' },
+      ...opts,
+    });
+    this._isLogEnabled = isLogEnabled;
+
+    if (this._isLogEnabled) {
+      this._httpClient.interceptors.response.use(
+        interceptors.logRequest.onResponse,
+        interceptors.logRequest.onError
+      );
+    }
+  }
+
+  async sendPromise<T>(url: string, jsonRequestData?: T): Promise<unknown> {
+    if (this._isLogEnabled) {
+      console.log(
+        'Sending request:',
+        url,
+        jsonRequestData && stringify(jsonRequestData).slice(0, LOGMAXLEN)
+      );
+    }
+
+    const response = await this._httpClient.request({
+      url,
+      method: jsonRequestData ? 'POST' : 'GET',
+      data: jsonRequestData,
+    });
+
+    return response?.data as T;
+  }
```

Comment:
@jurajpiar thank you for the changes, but I'd keep the requestData as `unknown` and the return data as typed with `T`. That would allow us to specify the expected type in axios.request<T>.
Furthermore, I'm not sure if we need to use `response?.data`; I think that response should be always defined. Shouldn't it?
```suggestion
  async sendPromise<T>(url: string, jsonRequestData?: unknown): Promise<T> {
    if (this._isLogEnabled) {
      console.log(
        'Sending request:',
        url,
        jsonRequestData && stringify(jsonRequestData).slice(0, LOGMAXLEN)
      );
    }

    const response = await this._httpClient.request<T>({
      url,
      method: jsonRequestData ? 'POST' : 'GET',
      data: jsonRequestData,
    });

    return response.data;
  }
```

Reply:
haha I've done it the other way around :D didn't even notice...

Reply:
done